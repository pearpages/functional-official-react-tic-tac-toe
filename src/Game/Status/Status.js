import React from 'react';
import PropTypes from 'prop-types';

import { getStatusMessage } from './status.helpers';

export function Status({ nextPlayer, winner }) {
  return (
    <div className="status">{getStatusMessage({ nextPlayer, winner })}</div>
  );
}

Status.propTypes = {
  nextPlayer: PropTypes.string.isRequired,
  winner: PropTypes.string,
};
