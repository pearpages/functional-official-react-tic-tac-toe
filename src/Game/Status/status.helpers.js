export function getStatusMessage({ nextPlayer, winner }) {
  if ((winner || '').toLowerCase() === 'draw') {
    return winner;
  } else if (winner) {
    return `Winner: ${winner}`;
  }
  return getNextPlayerMessage(nextPlayer);
}

export function getNextPlayerMessage(nextPlayer) {
  return `Next player: ${nextPlayer}`;
}
