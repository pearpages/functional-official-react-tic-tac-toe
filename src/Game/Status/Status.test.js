import { Status } from './Status';

import { render } from '@testing-library/react';

describe('<Status>', () => {
  test('Check when the html rendered changes', () => {
    const result = render(<Status nextPlayer={'X'} />);
    expect(result.container.innerHTML).toMatchSnapshot();
  });
  test('Winner', () => {
    const result = render(<Status nextPlayer={'O'} winner={'X'} />);
    expect(result.container.textContent).toBe('Winner: X');
  });
  test('Next player', () => {
    const result = render(<Status nextPlayer={'O'} />);
    expect(result.container.textContent).toBe('Next player: O');
  });
  test('draw', () => {
    const result = render(<Status nextPlayer={'O'} winner={'draw'} />);
    expect(result.container.textContent).toBe('draw');
  });
});
