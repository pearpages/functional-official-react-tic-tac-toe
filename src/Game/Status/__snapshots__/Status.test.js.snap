// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status> Check when the html rendered changes 1`] = `"<div class=\\"status\\">Next player: X</div>"`;
